engine/engine.go

```diff
@@ -416,7 +418,17 @@ func (r *ruleEngine) createViolation(conditionResponse ConditionResponse, rule R
 			incident.Message = templateString
 		}
 
-		incidents = append(incidents, incident)
+		// Checking duplication of incident
+
+		jsonIncident, _ := json.Marshal(incident)  // Converting to JSON to be used as keys of a map
```

Comment:
I am worried about the cost of this. Can we make a key function that only takes specific values to compare? 

Think like file URI, Message, and Linenumber if it exists in extras. I think that the cost of marshal could become too great, and the things that this is capturing as different issues, I think is misleading. 

For instance, if the type is a method or a local variable, as long as we are pointing to the same location, it is finding the same violation, just in a different search mechanism, and I think the user can be spared this detail IMO.